ports/minimal/uart_core.c

```diff
@@ -29,10 +29,14 @@ int mp_hal_stdin_rx_chr(void) {
 }
 
 // Send string of given length
-void mp_hal_stdout_tx_strn(const char *str, mp_uint_t len) {
+mp_uint_t mp_hal_stdout_tx_strn(const char *str, mp_uint_t len) {
+    mp_uint_t ret = len;
     #if MICROPY_MIN_USE_STDOUT
     int r = write(STDOUT_FILENO, str, len);
-    (void)r;
+    if (r >= 0) {
+        // in case of an error in the syscall, pretend write succeeded
```

Comment:
I think it's better to attempt to signal failure to the caller on error, same as elsewhere. This could be by returning zero bytes or raising an exception (I think raising an exception is what CPython will do in this case, but probably OK to go with zero for "minimal" size.)

Reply:
I agree, let's return zero for now and not add exceptions in the mix.